ballerina/natives.bal

```diff
@@ -148,10 +186,32 @@ isolated function print(string logLevel, string msg, error? err = (), *KeyValues
             logRecord[k] = v;
         }
     }
+    string logOutput = "";
     if format == "json" {
-        println(stderrStream(), java:fromString(logRecord.toJsonString()));
+        logOutput = logRecord.toJsonString();
     } else {
-        println(stderrStream(), java:fromString(printLogFmt(logRecord)));
+        logOutput = printLogFmt(logRecord);
+    }
+    lock {
+        if outputFilePath != "" {
+            fileWrite(logOutput);
+        } else {
+            println(stderrStream(), java:fromString(logOutput));
+        }
+    }
+}
+
+isolated function fileWrite(string logOutput) {
+    string output = logOutput;
+    lock {
+        string|io:Error content = io:fileReadString(outputFilePath);
+        if content != "" {
+            output = "\n" + output;
+        }
+        io:Error? result = io:fileWriteString(outputFilePath, output, io:APPEND);
```

Comment:
```suggestion
        io:Error? result = io:fileWriteString(outputFilePath, output + "\n", io:APPEND);
```

Reply:
we can avoid read operation. try to minimize the operations otherwise, it impact the performance

Reply:
If the file already has content, this would start writing from the tail of the last line of that file instead of starting from the next new line.

Reply:
if we append a new line at the end when writing. the new message will write to the new line. reading the content every time we write a message will impact performance a lot.

Reply:
Ack. Updated in 98b4983